### DISCUSSION TITLE
Should we have spell-check for quiz questions?

### DISCUSSION THREAD

**Discussion by a2937** [freeCodeCamp/Developer_Quiz_Site#627] [Ideas]
Several months ago on a project , I set up private dictionary files that were used to check the spelling of words. Would something like that be suitable here; provided specialized computing terms were added to a separate file? Each dictionary file in that project has a single word per line and all words are lower-case. While not a true grammar parser, it could be useful right?

**Comment by jdwilkin4** [comment]
I think a good option would be to create a precommit hook that runs a spell checker on all staged files.
This article has a good solution for that 
https://saransh.dev/blog/platform/automated-spellchecker-using-husky

It leverages this existing npm package
https://www.npmjs.com/package/spellchecker-cli

**Comment by a2937** [comment]
We will probably still need to add at least one custom dictionary for weirdly named variables and important people. But I do like that package a lot.

**Comment by a2937** [comment]
I just noticed some typos. 

https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/7847cb838188a1f1cae3f299d5db46d2f677571a/src/data/javascript-quiz.ts#L140

The following all the same typo twice. 

https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/7847cb838188a1f1cae3f299d5db46d2f677571a/src/data/linux-quiz.ts#L1070

https://github.com/freeCodeCamp/Developer_Quiz_Site/blob/7847cb838188a1f1cae3f299d5db46d2f677571a/src/data/linux-quiz.ts#L1059

**Reply by jdwilkin4** [reply]
Any typos you find can be addressed in a new PR 👍
